### PR TITLE
CTLungAnalyzer added

### PR DESCRIPTION
**Pull request by rbumm**
Thank you for contributing to 3D Slicer!

- **To add a new extension with this pull request**: Please keep content of "TODO list for submitting a new extension" section and put an 'x' character in the brackets for each todo item to indicate that you have accomplished that prerequisite.

- **To update an existing extension with this pull request**: Please delete all text in this template and just describe which extension is updated and optionally tell us in a sentence what has been changed. To make extension updates easier in the future (so that you don't have to submit a new pull request after each change in your extension), you may consider replacing specific git hash in your s4ext file by a branch name (for example: `master` or `release` for nightly version; `(majorVersion).(minorVersion)` such as `4.10` for stable Slicer version).

# TODO list for submitting a new extension

[To make sure users can find your extension, understand what it is intended for and how to use it, please complete the checklist below. You do not need to complete all the item by the time you submit the pull request, but most likely the changes will only be merged if all the tasks are done. See more information about the submission process here: https://slicer.readthedocs.io/en/latest/developer_guide/extensions.html]

- [ x] Extension has a reasonable name (not too general, not too narrow, suggests what the extension is for)
- [ x] Repository name is Slicer+ExtensionName
- [ x] Repository is associated with `3d-slicer-extension` GitHub topic so that it is listed [here](https://github.com/topics/3d-slicer-extension). To edit topics, click the settings icon in the right side of "About" section header and enter `3d-slicer-extension` in "Topics" and click "Save changes". To learn more about topics, read https://help.github.com/en/articles/about-topics
- [ x] Extension description summarizes in 1-2 sentences what the extension is usable (should be understandable for non-experts)
- [x ] Any known related patents must be mentioned in the extension description.
- [ x] If source code license is more restrictive for users than BSD, MIT, Apache, or 3D Slicer license then the name of the used license must be mentioned in the extension description.
- [ x] Extension URL and revision (scmurl, scmrevision) is correct, consider using a branch name (master, release, ...) instead of a specific git has to avoid re-submitting pull request whenever the extension is updated
- [ x] Extension icon URL is correct
- [ x] Screenshot URLs (screenshoturls) are correct, contains at least one
- [ x] Homepage URL points to valid webpage containing the following:
  - [x ] Extension name
  - [ x] Short description: 1-2 sentences, which summarizes what the extension is usable for
  - [ x] At least one nice, informative image, that illustrates what the extension can do. It may be a screenshot.
  - [ x] Description of contained modules: at one sentence for each module
  - [ x] Tutorial: step-by-step description of at least the most typical use case, include a few screenshots, provide download links to sample input data set
  - [ x] Publication: link to publication and/or to PubMed reference (if available)
  - [ x] License: We suggest you use a permissive license that includes patent and contribution clauses.  This will help protect developers and ensure the code remains freely available.  We suggest you use the [Slicer License](https://github.com/Slicer/Slicer/blob/master/License.txt) or the [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0). Always mention in your README file the license you have chosen.  If you choose a different license, explain why to the extension maintainers. Depending on the license we may not be able to host your work. Read [here](https://opensource.guide/legal/#which-open-source-license-is-appropriate-for-my-project) to learn more about licenses.
  - [ x] Content of submitted s4ext file is consistent with the top-level CMakeLists.txt file in the repository (description, URLs, dependencies, etc. are the same)

[If you have any questions or comments then please describe them here.]

### CODE REVIEW

**Comment by lassoan** [comment]
Thank you for your contribution. This is a very nice tool for lung assessment. To make sure that the module works well and prospective users can find this extension and figure out how to use it correctly, the followings would need to be addressed:
- [ ] Input segmentation node selector is missing, in general it is not safe to assume that the first segmentation node is the right one
- [ ] Input segment names are not matched robustly. It is assumed that first segment is right, second segment is left lung, but this is not safe enough in general. I would recommend to use take advantage of stadard terminology support built into the segment editor. If you double-click in the colored box in the segment list in segment editor then you can choose what the segment contains. You can type "lung" there and then choose "left" or "right" modifier. This sets the segment name but also saves this information using standard DICOM terminology codes, so this is a very robust way of specifying what a segment contains. In the analysis module, you could just check for the "left lung" and "right lung" segment names (and return with an error if no such segments are found).
- [ ] It is very hard to find usage instructions and hard to follow the steps described in one big paragraph. Rename "Methods" section to something like "Instructions" and break it down into a numbered list.
- [ ] Provide a tutorial that gives step-by-step instructions for downloading sample data (preferably using Sample Data module, see below), interact with the module, and show a screenshot of the expected output (so that the user can compare that he got the same result)
- [ ] Provide a sample data set (CT image and lung segmentation) that you and prospective users can use to try your module. It can be also used for automatic testing/validation of the module (compare with verified values). The skeleton for registering custom data sets is already there, you just need to replace the URL. I have healthy lung CT data set, but it would be much more representative to do it with a COVID patient case.
- [ ] Mention that lungs can be segmented fully automatically segmented from CT using the NvidiaAIAssistedAnnotation extension, with freely available lung CT segmentation models.
- [ ] In README.md you write "The code presented here is distributed under the https://www.apache.org/licenses/LICENSE-2.0 license.", but LICENSE file states that it is GPL. It would be great if you could replace the LICENSE file with the correct Apache license to avoid any confusions.
- [ ] Icon URL `https://github.com/rbumm/SlicerCTLungAnalyzer/blob/main/CTLungAnalyzer/Resources/Icons/CTLungAnalyzer.png` is not correct, it is the website URL that contains the icon; you can get the correct one from the "Download" button (https://github.com/rbumm/SlicerCTLungAnalyzer/raw/main/CTLungAnalyzer/Resources/Icons/CTLungAnalyzer.png)
- [ ] Line 35: `status` keyword is missing
- [ ] Remove last 11 lines that are included by accident (© 2020 GitHub, Inc. ... Blog)
- [ ] Remove this file, probably not needed: https://github.com/rbumm/SlicerCTLungAnalyzer/blob/main/CTLungAnalyzer/Resources/UI/CTLungAnalyzer.ui.nocholder
- [ ] Category: "Lung" is a bit too specific. We already have a "Chest Imaging Platform" category, so it would be better if this extension could be there, too (we'll rename this category to "Chest imaging" when we upgrade the extension manager in 1-2 months)
- [ ] Finalize extension name: why is it called CTLungAnalyzer and not LungCTAnalyzer?
- [ ] Typo: `rund the module again`

**Comment by pieper** [comment]
This is a very cool and timely contribution 👍 

@rbumm do reach out on discourse.slicer.org if any of the steps aren't clear.

**Comment by rbumm** [comment]
> Thank you for your contribution. This is a very nice tool for lung assessment. To make sure that the module works well and prospective users can find this extension and figure out how to use it correctly, the followings would need to be addressed:
> 
Thank you !
I am now in the process of dealing with all your requests. Done means done on my computer :-) 
> * [ ]  Input segmentation node selector is missing, in general it is not safe to assume that the first segmentation node is the right one
Done
> * [ ]  Input segment names are not matched robustly. It is assumed that first segment is right, second segment is left lung, but this is not safe enough in general. I would recommend to use take advantage of stadard terminology support built into the segment editor. If you double-click in the colored box in the segment list in segment editor then you can choose what the segment contains. You can type "lung" there and then choose "left" or "right" modifier. This sets the segment name but also saves this information using standard DICOM terminology codes, so this is a very robust way of specifying what a segment contains. In the analysis module, you could just check for the "left lung" and "right lung" segment names (and return with an error if no such segments are found).

Done, I added two segment selector drop down fields, which are checked upon processing the results for presence and correct naming: "Right lung mask" and "Left lung mask". Just for the records: The simple process of how to find out how to add these drop down fields in the QT editor and NOT getting it grayed out (Signal process) took me about 6 hrs of working time. It really needs clear documentation here. Even when I knew about that processing it was soo hard to find in the QT editor (right below hidden tab)   

> * [ ]  It is very hard to find usage instructions and hard to follow the steps described in one big paragraph. Rename "Methods" section to something like "Instructions" and break it down into a numbered list.
Will do so

> * [ ]  Provide a tutorial that gives step-by-step instructions for downloading sample data (preferably using Sample Data module, see below), interact with the module, and show a screenshot of the expected output (so that the user can compare that he got the same result)
Will do so

> * [ ]  Provide a sample data set (CT image and lung segmentation) that you and prospective users can use to try your module. It can be also used for automatic testing/validation of the module (compare with verified values). The skeleton for registering custom data sets is already there, you just need to replace the URL. I have healthy lung CT data set, but it would be much more representative to do it with a COVID patient case.

I have a dataset here, but do not know how to fully anonymized that (nrrd is enough ?) and how to upload it with a checksum. 
Secondly, I would need to have not only the Chest CT available for download but also the lung mask segmentation. How to do all that ? Here I really need help ! 

> * [ ]  Mention that lungs can be segmented fully automatically segmented from CT using the NvidiaAIAssistedAnnotation extension, with freely available lung CT segmentation models.

Will do so

> * [ ]  In README.md you write "The code presented here is distributed under the https://www.apache.org/licenses/LICENSE-2.0 license.", but LICENSE file states that it is GPL. It would be great if you could replace the LICENSE file with the correct Apache license to avoid any confusions.

ok

> * [ ]  Icon URL `https://github.com/rbumm/SlicerCTLungAnalyzer/blob/main/CTLungAnalyzer/Resources/Icons/CTLungAnalyzer.png` is not correct, it is the website URL that contains the icon; you can get the correct one from the "Download" button (https://github.com/rbumm/SlicerCTLungAnalyzer/raw/main/CTLungAnalyzer/Resources/Icons/CTLungAnalyzer.png)

Will look into this

> * [ ]  Line 35: `status` keyword is missing

I can not find this and how should I set the status ? 

> * [ ]  Remove last 11 lines that are included by accident (© 2020 GitHub, Inc. ... Blog)

I can not find this - where should I remove that ? Github, source code or website ? 

> * [ ]  Remove this file, probably not needed: https://github.com/rbumm/SlicerCTLungAnalyzer/blob/main/CTLungAnalyzer/Resources/UI/CTLungAnalyzer.ui.nocholder

Oh yes, definitely not needed. 
As I am not so familiar with GitHub - how do I technically remove files here ? I do not find a delete option in my project. 

> * [ ]  Category: "Lung" is a bit too specific. We already have a "Chest Imaging Platform" category, so it would be better if this extension could be there, too (we'll rename this category to "Chest imaging" when we upgrade the extension manager in 1-2 months)

Is done. 

> * [ ]  Finalize extension name: why is it called CTLungAnalyzer and not LungCTAnalyzer?

will be done. 

> * [ ]  Typo: `rund the module again`
> 
no longer needed, because function was deleted. 

> Let us know if you need help with addressing any of these comments.

Best regards
rudolf

File: CTLungAnalyzer.s4ext [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+# One line stating what the module does
+description This extension performs segmentation and 2D / 3D display of a chest CT into non-ventilated, ventilated, infiltrated and collapsed lung tissue as well as vessels. 
+
+# Space separated list of urls
+screenshoturls https://user-images.githubusercontent.com/18140094/98554410-f5ddd600-22a0-11eb-9196-b9223c8ada3f.jpg
+
+# 0 or 1: Define if the extension should be enabled after its installation.
+enabled 1
+© 2020 GitHub, Inc.
```
Comment (jamesobutler) [inline CTLungAnalyzer.s4ext:45]: > Remove last 11 lines that are included by accident

This line and all the ones below are the ones to remove.

**Comment by rbumm** [comment]
> * [ ]  Mention that lungs can be segmented fully automatically segmented from CT using the NvidiaAIAssistedAnnotation extension, with freely available lung CT segmentation models.

Tried the Nvidea AIAA (you mean the one in the segment editor?) again, but can only find presets for liver and spleen segmentation. Am I missing something here ? Thanks ...

**Comment by lassoan** [comment]
> I have a dataset here, but do not know how to fully anonymized that (nrrd is enough ?) and how to upload it with a checksum.
Secondly, I would need to have not only the Chest CT available for download but also the lung mask segmentation. How to do all that ? Here I really need help !

See instruction for generating hash here: https://github.com/rbumm/SlicerCTLungAnalyzer/blob/29a9bcd082f87f2d4b1689b3966cc883e4188ce5/CTLungAnalyzer/CTLungAnalyzer.py#L64-L65

You don't have to use the hash as filename, it is just one way of versioning. You can just use human-readable filename, and if you add new versions of the testing files in the future then use different filenames for those (...2.nrrd, ...3.nrrd).

You can create a new release on github (for example `TestingData`) and upload the image (.nrrd) and segmentation (.seg.nrrd) files there as assets in that release. After this, if you need help with creating the `registerCustomSampleDataSource` command then let me know.

> changed to GPL

Oh no, why? You cannot expect to get any benefit from using such license, but it is guaranteed to scare away potential collaborators/contributors. Very few people would want to contribute to a GPL-licensed project, because it would mean that they would invest time into a project that then they could not use any way they want. Please reconsider. Apache, BSD, Slicer license, etc. are all good. I would promote and help with this project (in the hope that it will help COVID patients now and in the future) but if it is hindered by GPL then I would not touch it.

**Comment by rbumm** [comment]
> > I have a dataset here, but do not know how to fully anonymized that (nrrd is enough ?) and how to upload it with a checksum.
> > Secondly, I would need to have not only the Chest CT available for download but also the lung mask segmentation. How to do all that ? Here I really need help !
> 
> See instruction for generating hash here: https://github.com/rbumm/SlicerCTLungAnalyzer/blob/29a9bcd082f87f2d4b1689b3966cc883e4188ce5/CTLungAnalyzer/CTLungAnalyzer.py#L64-L65
> 

found that

> You don't have to use the hash as filename, it is just one way of versioning. You can just use human-readable filename, and if you add new versions of the testing files in the future then use different filenames for those (...2.nrrd, ...3.nrrd).

ok 

> 
> You can create a new release on github (for example `TestingData`) and upload the image (.nrrd) and segmentation (.seg.nrrd) files there as assets in that release. After this, if you need help with creating the `registerCustomSampleDataSource` command then let me know.
> 

ok

> > changed to GPL
> 
> Oh no, why? You cannot expect to get any benefit from using such license, but it is guaranteed to scare away potential collaborators/contributors. Very few people would want to contribute to a GPL-licensed project, because it would mean that they would invest time into a project that then they could not use any way they want. Please reconsider. Apache, BSD, Slicer license, etc. are all good. I would promote and help with this project (in the hope that it will help COVID patients now and in the future) but if it is hindered by GPL then I would not touch it.

is now changed to Apache license :-)

**Comment by rbumm** [comment]
> > Secondly, I would need to have not only the Chest CT available for download but also the lung mask segmentation. How to do all that ? Here I really need help !
> 
> See instruction for generating hash here: https://github.com/rbumm/SlicerCTLungAnalyzer/blob/29a9bcd082f87f2d4b1689b3966cc883e4188ce5/CTLungAnalyzer/CTLungAnalyzer.py#L64-L65
> 
> You don't have to use the hash as filename, it is just one way of versioning. You can just use human-readable filename, and if you add new versions of the testing files in the future then use different filenames for those (...2.nrrd, ...3.nrrd).
> 
> You can create a new release on github (for example `TestingData`) and upload the image (.nrrd) and segmentation (.seg.nrrd) files there as assets in that release. After this, if you need help with creating the `registerCustomSampleDataSource` command then let me know.
> 

As expected, I run into problems.
And it is only the upload of 3 files :-(

What I did: 
Opened Slicer, created two nrrd datasets from the chest CT. 

Saved them in a sep. dir and created the SHA256 hash. 
Forked the Slicer/SlicerTestingData to my Github. 
Cloned that own forked Project to my Laptop. 
Installed Python in my laptop.
Tried to run process_release_data.py with python
Ran into the 'no module named github_release' problem you are certainly familiar with. 
Solved that by  'python -m pip install githubrelease'
'no module named github_release' problem disappeared
but  after the command 
'python process_release_data.py upload --github-token 123123...123'
a new directory is created in my SlicerTestingData directory containing new folder SHA256 and the two nrrd files with hashed file names and the SH256.csv file  - but the script ends with an error: 

C:\Users\Rudolf\Documents\GitHub\SlicerTestingData> python process_release_data.py upload
release SHA256: downloading SHA256.csv
Traceback (most recent call last):
  File "C:\Users\Rudolf\Documents\GitHub\SlicerTestingData\process_release_data.py", line 382, in <module>
    upload(repo_name, root_dir, incoming_dir, hashalgo, github_token)
  File "C:\Users\Rudolf\Documents\GitHub\SlicerTestingData\process_release_data.py", line 317, in upload
    github_release.gh_release_create(repo_name, hashalgo, publish=True)
  File "C:\Users\Rudolf\AppData\Local\Programs\Python\Python39\lib\site-packages\github_release.py", line 111, in with_check_for_credentials
    raise EnvironmentError(
OSError: This command requires credentials provided by passing --github-token CLI argument, set using GITHUB_TOKEN env. variable or using netrc file. For more details, see https://github.com/j0057/github-release#configuring
PS C:\Users\Rudolf\Documents\GitHub\SlicerTestingData> python process_release_data.py upload --github-token 123123...123
Traceback (most recent call last):
  File "C:\Users\Rudolf\Documents\GitHub\SlicerTestingData\process_release_data.py", line 382, in <module>
    upload(repo_name, root_dir, incoming_dir, hashalgo, github_token)
  File "C:\Users\Rudolf\Documents\GitHub\SlicerTestingData\process_release_data.py", line 263, in upload
    hashalgo_csv = download_fileindex_csv(
  File "C:\Users\Rudolf\Documents\GitHub\SlicerTestingData\process_release_data.py", line 77, in download_fileindex_csv
    if not github_release.gh_asset_download(repo_name, hashalgo, hashalgo + ".csv"):
  File "C:\Users\Rudolf\AppData\Local\Programs\Python\Python39\lib\site-packages\github_release.py", line 866, in gh_asset_download
    releases = get_releases(repo_name)
  File "C:\Users\Rudolf\AppData\Local\Programs\Python\Python39\lib\site-packages\github_release.py", line 249, in get_releases
    _recursive_gh_get(
  File "C:\Users\Rudolf\AppData\Local\Programs\Python\Python39\lib\site-packages\github_release.py", line 155, in _recursive_gh_get
    response.raise_for_status()
  File "C:\Users\Rudolf\AppData\Local\Programs\Python\Python39\lib\site-packages\requests\models.py", line 943, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 401 Client Error: Unauthorized for url: https://api.github.com/repos/Slicer/SlicerTestingData/releases
PS C:\Users\Rudolf\Documents\GitHub\SlicerTestingData> python process_release_data.py upload --github-token 123123...123
Traceback (most recent call last):
  File "C:\Users\Rudolf\Documents\GitHub\SlicerTestingData\process_release_data.py", line 382, in <module>
    upload(repo_name, root_dir, incoming_dir, hashalgo, github_token)
  File "C:\Users\Rudolf\Documents\GitHub\SlicerTestingData\process_release_data.py", line 263, in upload
    hashalgo_csv = download_fileindex_csv(
  File "C:\Users\Rudolf\Documents\GitHub\SlicerTestingData\process_release_data.py", line 77, in download_fileindex_csv
    if not github_release.gh_asset_download(repo_name, hashalgo, hashalgo + ".csv"):
  File "C:\Users\Rudolf\AppData\Local\Programs\Python\Python39\lib\site-packages\github_release.py", line 866, in gh_asset_download
    releases = get_releases(repo_name)
  File "C:\Users\Rudolf\AppData\Local\Programs\Python\Python39\lib\site-packages\github_release.py", line 249, in get_releases
    _recursive_gh_get(
  File "C:\Users\Rudolf\AppData\Local\Programs\Python\Python39\lib\site-packages\github_release.py", line 155, in _recursive_gh_get
    response.raise_for_status()
  File "C:\Users\Rudolf\AppData\Local\Programs\Python\Python39\lib\site-packages\requests\models.py", line 943, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 401 Client Error: Unauthorized for url: https://api.github.com/repos/Slicer/SlicerTestingData/releases

Thanks for any help.

**Comment by rbumm** [comment]
> Thank you for your contribution. This is a very nice tool for lung assessment. To make sure that the module works well and prospective users can find this extension and figure out how to use it correctly, the followings would need to be addressed:
> 
> * [ ]  Input segmentation node selector is missing, in general it is not safe to assume that the first segmentation node is the right one
> * [ ]  Input segment names are not matched robustly. It is assumed that first segment is right, second segment is left lung, but this is not safe enough in general. I would recommend to use take advantage of stadard terminology support built into the segment editor. If you double-click in the colored box in the segment list in segment editor then you can choose what the segment contains. You can type "lung" there and then choose "left" or "right" modifier. This sets the segment name but also saves this information using standard DICOM terminology codes, so this is a very robust way of specifying what a segment contains. In the analysis module, you could just check for the "left lung" and "right lung" segment names (and return with an error if no such segments are found).
> * [ ]  It is very hard to find usage instructions and hard to follow the steps described in one big paragraph. Rename "Methods" section to something like "Instructions" and break it down into a numbered list.
> * [ ]  Provide a tutorial that gives step-by-step instructions for downloading sample data (preferably using Sample Data module, see below), interact with the module, and show a screenshot of the expected output (so that the user can compare that he got the same result)
> * [ ]  Provide a sample data set (CT image and lung segmentation) that you and prospective users can use to try your module. It can be also used for automatic testing/validation of the module (compare with verified values). The skeleton for registering custom data sets is already there, you just need to replace the URL. I have healthy lung CT data set, but it would be much more representative to do it with a COVID patient case.
> * [ ]  Mention that lungs can be segmented fully automatically segmented from CT using the NvidiaAIAssistedAnnotation extension, with freely available lung CT segmentation models.
> * [ ]  In README.md you write "The code presented here is distributed under the https://www.apache.org/licenses/LICENSE-2.0 license.", but LICENSE file states that it is GPL. It would be great if you could replace the LICENSE file with the correct Apache license to avoid any confusions.
> * [ ]  Icon URL `https://github.com/rbumm/SlicerCTLungAnalyzer/blob/main/CTLungAnalyzer/Resources/Icons/CTLungAnalyzer.png` is not correct, it is the website URL that contains the icon; you can get the correct one from the "Download" button (https://github.com/rbumm/SlicerCTLungAnalyzer/raw/main/CTLungAnalyzer/Resources/Icons/CTLungAnalyzer.png)
> * [ ]  Line 35: `status` keyword is missing
> * [ ]  Remove last 11 lines that are included by accident (© 2020 GitHub, Inc. ... Blog)
> * [ ]  Remove this file, probably not needed: https://github.com/rbumm/SlicerCTLungAnalyzer/blob/main/CTLungAnalyzer/Resources/UI/CTLungAnalyzer.ui.nocholder
> * [ ]  Category: "Lung" is a bit too specific. We already have a "Chest Imaging Platform" category, so it would be better if this extension could be there, too (we'll rename this category to "Chest imaging" when we upgrade the extension manager in 1-2 months)
> * [ ]  Finalize extension name: why is it called CTLungAnalyzer and not LungCTAnalyzer?
> * [ ]  Typo: `rund the module again`
> 
> Let us know if you need help with addressing any of these comments.

I tried to resolve all points mentioned, implemented a test routine, and hosted the COVID demo data on a personal server. 
Would be great if you could do another check and hopefully include the extension.
Thank you, 
Best regards
r

File: CTLungAnalyzer.s4ext [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+# Firstname1 Lastname1 ([SubOrg1, ]Org1), Firstname2 Lastname2 ([SubOrg2, ]Org2)
+# For example: Jane Roe (Superware), John Doe (Lab1, Nowhere), Joe Bloggs (Noware)
+contributors Rudolf Bumm (KSGR)
+
+# Match category in the xml description of the module (where it shows up in Modules menu)
+category Chest Imaging Platform
+
+# url to icon (png, size 128x128 pixels)
+iconurl https://github.com/rbumm/SlicerCTLungAnalyzer/raw/main/CTLungAnalyzer/Resources/Icons/CTLungAnalyzer.png
```
Comment (pieper) [inline CTLungAnalyzer.s4ext:31]: I get a 404 for this url

**Review by pieper — APPROVED** [review]
LGTM - just one broken link to the URL.  Thanks for your persistence!

@lassoan @jamesobutler any other suggestions?

**Comment by rbumm** [comment]
> LGTM - just one broken link to the URL. Thanks for your persistence!
> 
> @lassoan @jamesobutler any other suggestions?

ICON URL is now changed.

File: CTLungAnalyzer.s4ext [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+# Match category in the xml description of the module (where it shows up in Modules menu)
+category Chest Imaging Platform
+
+# url to icon (png, size 128x128 pixels)
+iconurl https://github.com/rbumm/SlicerCTLungAnalyzer/raw/master/CTLungAnalyzer/Resources/Icons/CTLungAnalyzer.png
+
+# Give people an idea what to expect from this code
+#  - Is it just a test or something you stand behind?
+Beta
```
Comment (jamesobutler) [inline CTLungAnalyzer.s4ext:35]: I believe this line should be formatted as `status Beta`

File: CTLungAnalyzer.s4ext [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+scmurl git://github.com/rbumm/SlicerCTLungAnalyzer
+scmrevision master
+
+# list dependencies
+# - These should be names of other modules that have .s4ext files
+# - The dependencies will be built first
+depends NA
+
+# Inner build directory (default is ".")
+build_subdirectory inner-build
```
Comment (jamesobutler) [inline CTLungAnalyzer.s4ext:18]: Your extension doesn’t appear to build as a Superbuild so this line should be `build_subdirectory .`

File: CTLungAnalyzer.s4ext [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+scmurl git://github.com/rbumm/SlicerCTLungAnalyzer
+scmrevision master
+
+# list dependencies
+# - These should be names of other modules that have .s4ext files
+# - The dependencies will be built first
+depends NA
+
+# Inner build directory (default is ".")
+build_subdirectory
```
Comment (jamesobutler) [inline CTLungAnalyzer.s4ext:18]: this line should be `build_subdirectory .`

**Comment by jamesobutler** [comment]
@rbumm I posted https://github.com/rbumm/SlicerCTLungAnalyzer/pull/4 which is necessary to make sure the extension can configure successfully as part of the build process.

**Review by jamesobutler — APPROVED** [review]
@lassoan Feel free to review the changes that were made based on your detailed review.

**Comment by rbumm** [comment]
> @lassoan Feel free to review the changes that were made based on your detailed review.

To understand the process better: 
(1) I have two approvals now before the merge, and need a third one ? 
(2) Once I am merged and need to do an update, will it again need reviews for the commit by three reviewers ? 

Thanks, have a great day
rudolf

**Comment by pieper** [comment]
Since we didn't hear anything we'll take that as a passive okay (or lack of time to review).  Thanks for your patience!

Next step is to watch tomorrow's dashboard for any build issues and tease our corrective actions if needed.  Since the build is pointed to your master branch you can be in control of this process : )

**Comment by rbumm** [comment]
> Since we didn't hear anything we'll take that as a passive okay (or lack of time to review). Thanks for your patience!
> 
> Next step is to watch tomorrow's dashboard for any build issues and tease our corrective actions if needed. Since the build is pointed to your master branch you can be in control of this process : )

That sounds great, thank you. Future changes on my side (f.e. I added a better extension test today, need to commit that later) will then automatically be in the Slicer builds ?

**Comment by pieper** [comment]
> Future changes on my side (f.e. I added a better extension test today, need to commit that later) will then automatically be in the Slicer builds ?

Yes, exactly - you can check the [dashboard](http://slicer.cdash.org/index.php?project=SlicerPreview) to see issues on the three platforms based on the master branch as of build time (nighttime in the US).  Reach out if you see things that don't make sense.

**Comment by lassoan** [comment]
I've spent some time revamping the extension, simplifying the code and GUI and adding real-time volume rendering, and make the GUI better follow Slicer conventions. I'll send a pull request with the changes today.

**Comment by lassoan** [comment]
Pull request sent, please merge: https://github.com/rbumm/SlicerCTLungAnalyzer/pull/5

I've managed to implement immediate updated of the color categorization, both in 2D and 3D (using colormaps and volume rendering): **https://youtu.be/0plsoy94hFE**
